src/molecules/top-navigation-bar.tsx

```diff
@@ -1,5 +1,9 @@
+"use client";
+
+import React, { useState } from 'react';
 import styles from "./top-navigation-bar.module.css";
 import { MdMenu } from "react-icons/md";
+import SideMenu from "../molecules/side-menu";
```

Comment:
```suggestion
import SideMenu, { SideMenuProps } from "../molecules/side-menu";
```
`SideMenuProps` の import が必要です.